broker-core/src/main/java/io/zeebe/broker/workflow/state/VariablesState.java

```diff
@@ -429,6 +429,10 @@ public void setListener(VariableListener listener) {
     this.listener = listener;
   }
 
+  public void resetListener() {
```

Comment:
If this is only for testing, let's make it package private, should accessible by the test as well. Or why not set the listener statically as a `RecordingVariableListener` and reset the listener instance (i.e. clear its lists) between every tests? I'd prefer either over a new public method that's only for testing.

Reply:
Agree which one do you prefer?

Reply:
I'd prefer the second one since it doesn't pollute the "production" code, but I'd approve either, package private methods for unit tests are fine as far as I'm concerned.

Reply:
Please see the new commit